lib/Resource.js

```diff
@@ -267,9 +267,15 @@ Resource.prototype.fetch = Promise.denodeify(function(relationName, callback) {
   var self = this;
   self._client._getRelated(this, relationName, function(err, newResources) {
     if (!newResources) return callback("No related resources found");
-    self._raw.relationships[relationName].data = newResources.map(function(someResource) {
-      return someResource._getBase();
-    });
+
+    if (!(newResources instanceof Array)) {
+      self._raw.relationships[relationName].data = newResources._getBase();
+    } else {
+      self._raw.relationships[relationName].data = newResources.map(function(someResource) {
+        return someResource._getBase();
+      });
+    }
```

Comment:
Can we do this in a _return early_ like above, avoiding the `else` block, which should be more readable.